react/DynamicIframe.tsx

```diff
@@ -15,15 +16,14 @@ const DynamicIframe: StorefrontFunctionComponent<DynamicIframeProps> = ({
   const src = dynamicSrc.replace(/({[A-z0-9]*})/g, function(match: string) {
     const thisParam = match.replace(/{|}/g, '')
     if (thisParam && params[thisParam]) {
-      return params[thisParam]
-    } else {
-      console.error(
-        'parameter ' + thisParam + ' not found in runtime params: ' + params
-      )
-      return ''
+      paramExists = true
```

Comment:
I think this should be reversed ... start with a true boolean, and if a bad param is found, set it to false. Currently if a bad param is in the mix but there's a good param after it, the boolean would end up true. Naming the boolean "allParamsExist" might be a better way to think about it. I'd also move the initialization of the var to line 16 (inside the functioncomponent).

Also when Breno recommended to "remove the else", he didn't mean that you should remove the code inside the else. You can keep the console.error after the if statement, but it doesn't need to be inside an else. Because the if statement contains a return, the code after the if will only be hit if the conditional is not met.

Reply:
thanks for the clarification on the else, but is there a downside to leaving it in place? In my opinion the presence of the else creates a more readable, explicitly defined structure rather than relying on the implicit assumption that the code below the if will be ignored should the return statement be run.

Reply:
To my knowledge there's no official policy at the moment, so you should be okay leaving it in. The front end devs are discussing things like that in #js-standards. It could be that a rule is added eventually, so I recommend participating in the votes there.